FHEM/00_SIGNALduino.pm

```diff
@@ -2187,95 +2187,112 @@ sub SIGNALduino_Parse_MU($$$$@)
 			my $bit_length = ($signal_length/((scalar @{$ProtocolListSIGNALduino{$id}{one}} + scalar @{$ProtocolListSIGNALduino{$id}{zero}})/2));
 			Debug "Expect $bit_length bits in message"  if ($valid && $debug);
 
-			#Check calculated min length
-			#$valid = $valid && $ProtocolListSIGNALduino{$id}{length_min} <= $bit_length if (exists $ProtocolListSIGNALduino{$id}{length_min}); 
-			#Check calculated max length
-			#$valid = $valid && $ProtocolListSIGNALduino{$id}{length_max} >= $bit_length if (exists $ProtocolListSIGNALduino{$id}{length_max});
-
-			#next if (!$valid) ;
-			#Debug "expecting $bit_length bits in signal" if ($debug && $valid);
-
 			Debug "Searching in patternList: ".Dumper(\%patternList) if($debug);
 			next if (!$valid) ;
 
-	
+			my @msgStartLst;
+			my $startStr="";
+			my $message_start=0 ;
+			my $startLogStr="";
+			
+			if (defined($ProtocolListSIGNALduino{$id}{start}))	# wenn start definiert ist, dann startStr ermitteln und in rawData suchen und in der rawData alles bis zum startStr abschneiden
```

Comment:
bislang wurde wir folgt abgefragt:

`if (@msgStartLst = SIGNALduino_getProtoProp($id,"start"))`

Würde das auch weiterhin funktionieren?

Reply:
@Ralf9 

Das war z.B. eine der offenen fragen.

Reply:
Meine Kommentare stehen auf pending, was bedeutet dies?

![github_pending1](https://user-images.githubusercontent.com/14318704/46501343-7f4e8e80-c825-11e8-991b-e0562cd922bb.png)

Reply:
SIGNALduino_getProtoProp nimmt drei Parameter (auch wenn nur zwei darüber beschrieben wurden. :( )

Wenn wir es so wie bisher aufrufen und start ist nicht definiert, dann kommt ein undef zurück.
`SIGNALduino_getProtoProp($id,"start")`

Die Variable `@msgStartLst ` bekommt dann den Wert `undef`. Leider wird der aber in den array eingefügt. :(
Hier steht das ganze verhalten gut erklärt und auch was wir nicht machen sollten :)
https://perlmaven.com/how-to-return-undef-from-a-function

vielleicht könnte man auch was über den 3. Parameter machen. Ich habe es aber auf Anhieb nicht geschafft.

Hier habe ich das Verhalten in einem kleinen Beispielprogramm geprüft:
https://onlinegdb.com/rJN8RzVqm

Zusammenfassend kann man sagen, weder die von mir vorgeschlagene noch die von dir vorgeschlagene Variante funktionieren wirklich.

Reply:
Mir ist nicht klar was damit nicht funktionieren soll?
```
if (defined($ProtocolListSIGNALduino{$id}{start}))	# wenn start definiert ist, dann startStr ermitteln und in rawData suchen und in der rawData alles bis zum startStr abschneiden
	{
		@msgStartLst = $ProtocolListSIGNALduino{$id}{start};
		if ( ($startStr=SIGNALduino_PatternExists($hash,@msgStartLst,\%patternList,\$rawData)) eq -1)
```

Reply:
Es prüft nicht ob eine Liste im Feld start steht, was eigentlich geprüft werden sollte. Mit einem Scalar Wert kommt es zu Fehlern im Programmablauf.

Reply:
Wäre es nicht besser, wenn dieser Syntax Test bereits im Travis Test gemacht würde damit es direkt nach einem commit auffällt?

Ich bin mit dem Fahrrad unterwegs und antworte erst heute Abend

Reply:
Ja, können wir dort prüfen, dafür brauchen wir eine Funktion die das pro Protokoll prüft.

Da ja auch immer wieder mal manuelle Änderungen der Protokoll Datei vorgenommen werden sollte der erweiterte Syntax Check beim Laden der Datei erfolgen .

Mit Travis können wir dann jedes Protokoll auf die richtigen Ergebnisse testen.